CSCFAIRMETA-684: [ADD] Integrate Etsin & Qvain with Fairdata SSO

MattiasLevlin opened the pull request:
- Integrate Etsin login with Fairdata SSO

MattiasLevlin commented:
To do:
- Set all required SSO session variables in authentication.py
- Account for different environments (test/stable/demo...) in variable names/functions

MattiasLevlin commented:
Correction to previous commit: - There are ida_projects and idm_groups, but no **ida_groups**

MattiasLevlin commented:
Functional now:
- Login/logout
- Retrieving datasets
- Publishing datasets

To do:
- Enable adding files from IDA
- Account for different environments (test/stable/demo...) in variable names/functions

Related [WIP] PR: https://github.com/CSCfi/etsin-ops/pull/31

MattiasLevlin commented:
Prerequisites:
https://github.com/CSCfi/etsin-ops/pull/32

For testing instructions, see:
https://wiki.eduuni.fi/display/cscetsin/Etsin%3A+SSO

Description:
- Integrates Etsin and Qvain with Fairdata SSO
- For now, both use Etsin authentication

What should be verified as functional:
- Login/logout
- Publishing and updates to datasets
- Adding files from IDA using v1 & v2 filepickers
- Deleting files and datasets

tahme reviewed (changes requested):
Looks good. Tested locally that login works, and when Etsin and Qvain Light run in separate `.local.fd-test.csc.fi` domains, logging into Etsin logs you also into Qvain as it's supposed to work. I have just a few minor cleanup suggestions.

I did notice that the relay parameter of the Etsin /sso endpoint is not working with SSO. It's the parameter that redirects you back to the page where you were after you login. That can be fixed in a separate issue.

MattiasLevlin commented on etsin_finder/frontend/js/components/dataset/events/index.jsx:
```diff
@@ -167,7 +167,7 @@ class Events extends Component {
 
     if (process.env.NODE_ENV === 'test') { /* test and stable */
-      url = `https://etsin-test.fairdata.fi/dataset/${identifier}`
+      url = `https://etsin.fd-test.csc.fi/dataset/${identifier}`
```
Addressed in 7845232

MattiasLevlin commented on etsin_finder/qvain_light_utils.py:
```diff
@@ -497,15 +478,15 @@ def check_if_data_in_user_IDA_project(data):
     # Check IDA project permissions, if user_ida_projects exist
     try:
-        user_ida_projects_ids = [project.split(":")[1] for project in user_ida_projects]
+        user_ida_projects_ids = [project for project in user_ida_projects]
```
Addressed in 84a398b

MattiasLevlin commented:
> I did notice that the relay parameter of the Etsin /sso endpoint is not working with SSO. It's the parameter that redirects you back to the page where you were after you login. That can be fixed in a separate issue.

Good catch. This can be addressed in the next PR (CSCFAIRMETA-685), which should separate the Qvain login from the Etsin one.

MattiasLevlin commented:
Commit 644c7ad requires https://github.com/CSCfi/etsin-ops/pull/33/files and addresses issues with logging in in test and demo, if the direct proxy is used

tahme commented on etsin_finder/frontend/js/components/general/loggedInUser.jsx:
```diff
@@ -7,14 +7,16 @@ import { faUser } from '@fortawesome/free-solid-svg-icons'
 const LoggedInUser = ({ Stores }) => {
   const { user, userLogged } = Stores.Auth
-  const { firstName } = user
+  const { firstName, lastName } = user
```
I think we should only show the first name. We initially tried to show both names but it took too much space. Or if it's possible that only one of them is defined, maybe show `firstName || lastName`.

tahme reviewed (approved):
:+1:
